dask/array/optimization.py

```diff
@@ -127,21 +156,21 @@ def optimize_slices(dsk):
     fancy_ind_types = (list, np.ndarray)
     dsk = dsk.copy()
     for k, v in dsk.items():
-        if type(v) is tuple and v[0] in GETTERS and len(v) in (3, 5):
+        if (get := _is_getter_task(v)) and len(v) in (3, 5):
             if len(v) == 3:
-                get, a, a_index = v
+                _, a, a_index = v
```

Comment:
I haven't thought this through that much, but loosely it feels like the 3/5 length logic and tuple decomposition could all be done in `_is_getter_task`, which could return a 5-tuple or `None` in the case it's not a getter. Would make the logic below simpler too, and I'd guess it'd be mildly faster too since the 3/5 check wouldn't happen twice.

Reply:
Good thought, I'll try it out to see if that simplifies things a bit

Reply:
Yeah, seemed to be mildly cleaner to me.